usl_pipeline/usl_lib/usl_lib/writers/polygon_writers.py

```diff
@@ -0,0 +1,40 @@
+import typing
+from typing import Tuple
+
+from shapely import geometry
+
+
+def num_list_to_strings(num_list):
+    return list(map(lambda v: str(v), num_list))
+
+
+def write_polygons_to_text_file(
+    polygons_with_masks: list[Tuple[geometry.Polygon, int]],
+    output_file: typing.TextIO,
+    support_mask_values: bool = False,
+) -> None:
+    """Writes polygon information with optional mask values to text file.
+
+    Args:
+        polygons_with_masks: List of tuples combining polygon with associated mask.
+        output_file: File text stream to write to.
+        support_mask_values: Optional indicator that masks should be added into output
+            as additional first column.
+
+    Returns:
+        The generator of tuples combining polygon with associated mask to iterate over.
+    """
+    output_file.write(f"{len(polygons_with_masks)}\n")
+    for polygon_mask in polygons_with_masks:
+        xx, yy = polygon_mask[0].exterior.coords.xy
+        x_list = xx.tolist()
+        y_list = yy.tolist()
+        mask_prefix = f"{polygon_mask[1]} " if support_mask_values else ""
```

Comment:
Similar to the previous PR, can we detect if a mask is present here versus asking the caller to pass  `support_mask_values`?

Reply:
This time it's not about detection and more like about preparing textual file with polygons that CityCat program can use. I want to be able to define precisely whether we want to have a column with masks in the polygon file or not (for instance for buildings we shouldn't have one and for green areas we may or may not depending on the soil info presence).